src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/LocalBuilder.cs

```diff
@@ -12,5 +12,21 @@ public abstract class LocalBuilder : LocalVariableInfo
         /// This constructor is invoked by derived classes.
         /// </remarks>
         protected LocalBuilder() { }
+
+        /// <summary>
+        /// Sets the name of this local variable.
+        /// </summary>
+        /// <param name="name">The name of the local variable</param>
+        /// <exception cref="InvalidOperationException">The containing type has been created with CreateType() or
+        /// containing type doesn't support symbol writing.</exception>"
+        public void SetLocalSymInfo(string name) => SetLocalSymInfoCore(name);
```

Comment:
Should we check for `null` here?

Reply:
It was not validated in .NET framework, and not documented that it throws ANE, will check if it throws with a sample code

Reply:
When null passed it throws on `TypeBuilder.CreateType()`, so I added validation here. Now the question is what should happen for locals that has no name, from the [specification:](https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#localvariable-table-0x33) `There shall be no duplicate rows in the LocalVariable table, based upon owner and Name.`. If we add a local with default string handle that could cause duplicate, so I updated the logic to not add the local into the table if the name was null or empty. NET framework doesn't require all locals to be named (though I cannot find a way to see PDB tables generated with netfx). Tag @tmat @hoyosjs  for opinions

Reply:
@buyaa-n You should be able to see these details using https://github.com/aaronrobinsonmsft/dnmd. Let me know if you need any help with that.

Reply:
> Let me know if you need any help with that.

Tried to build the project, 2nd step `cmake -S . -B artifacts` fails with:
```log
CMake Error in test/regpal/CMakeLists.txt:
  Imported target "nethost" includes non-existent path

    "C:\Program Files\dotnet\sdk\9.0.100-preview.1.24101.2\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(313,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [D:\dotnet\DNMD\test\FindNetHostDir.proj]
    C:\Program Files\dotnet\packs/Microsoft.NETCore.App.Host.win-x64/9.0.0-preview.1.24080.9/runtimes/win-x64/native"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

Reply:
Sad. Please comment out the following lines and try again.

https://github.com/AaronRobinsonMSFT/DNMD/blob/f6e4b6c246f5e94ae7adf9d934881130f65fec41/CMakeLists.txt#L20-L22

/cc @jkoritzinsky

Reply:
Build worked, thank you.

Reading a dll works, but PDB fails:
```
.\mddump.exe MyAssembly.pdb
Failed to read file as PE or metadata blob.
````

Reply:
Hmmmm. Can you share the PDB with me offline?

Reply:
Sure, I'll share tomorrow

Reply:
> so I updated the logic to not add the local into the table if the name was null or empty. NET framework doesn't require all locals to be named (though I cannot find a way to see PDB tables generated with netfx).

Update for the above comment: looks skipping the unnamed local variables is the way to go, now I see the same behavior for local variables that have set name with `SetLocalSymInfo` for portable PDBs and .NET framework PDB

Reply:
Nice job! Appreciate the diligence on this.